[csolution] IAR compiler architecture not found by CMake ?

With csolution 0.9.2 and these settings in IAR.8.50.6.cmake:
```
############### EDIT BELOW ###############
# Set base directory of toolchain
set(TOOLCHAIN_ROOT "/c/Program Files (x86)/IAR Systems/Embedded Workbench 8.4/arm/bin")
set(EXT .exe)
```

When building I hit an error:
```
$ cbuild.sh gpio-toggle.IAR+B-U585I-IOT02A.cprj
(cbuild.sh): Build Invocation 0.10.5 (C) 2021 ARM
gpio-toggle.IAR+B-U585I-IOT02A.cprj validates
(cbuildgen): Build Process Manager 0.10.5 (C) 2022 ARM
M650: Command completed successfully.
(cbuildgen): Build Process Manager 0.10.5 (C) 2022 ARM
M652: Generated file for project build: 'C:/Cube20/stm32cube_fw_all/examples/hal/gpio/toggle/projects/gpio-toggle.IAR+B-U585I-IOT02A_IntDir/CMakeLists.txt'
CMake Error at C:/Program Files/CMake/share/cmake-3.22/Modules/CMakeFindBinUtils.cmake:141 (message):
  Failed to find linker and librarian for IAR on .
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.22/Modules/CMakeDetermineCCompiler.cmake:196 (include)
  CMakeLists.txt:61 (project)
```

Looks like in CMake : 
  ${CMAKE_${_CMAKE_PROCESSING_LANGUAGE}_COMPILER_ARCHITECTURE_ID}
returns an empty string

Though CMakeLists.txt looks good:
```
# Toolchain config map

include ("C:/cmsis-toolbox/etc/IAR.8.50.6.cmake")

# Setup project

project(${TARGET} LANGUAGES C)
```

--- a/tools/buildmgr/cbuildgen/config/IAR.8.50.6.cmake
+++ b/tools/buildmgr/cbuildgen/config/IAR.8.50.6.cmake
@@ -135,10 +135,10 @@ set (LIB_SUFFIX ".a")
 set (EXE_SUFFIX ".out")
 
 # ELF to HEX conversion
-set (ELF2HEX --silent --ihex "${OUT_DIR}${TARGET}${EXE_SUFFIX}" "${OUT_DIR}${TARGET}.hex")
+set (ELF2HEX --silent --ihex "${OUT_DIR}/${TARGET}${EXE_SUFFIX}" "${OUT_DIR}/${TARGET}.hex")
 
 # ELF to BIN conversion
-set (ELF2BIN --silent --bin "${OUT_DIR}${TARGET}${EXE_SUFFIX}" "${OUT_DIR}${TARGET}.bin")
+set (ELF2BIN --silent --bin "${OUT_DIR}/${TARGET}${EXE_SUFFIX}" "${OUT_DIR}/${TARGET}.bin")
 
 # Set CMake variables for toolchain initialization
 set(CMAKE_SYSTEM_NAME Generic)
@@ -151,9 +151,6 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/CMakeASM")
 
 # Set CMake variables for skipping compiler identification
 set(CMAKE_ASM_COMPILER_FORCED TRUE)
-set(CMAKE_C_COMPILER_ID "IAR")
-set(CMAKE_C_COMPILER_ID_RUN TRUE)
-set(CMAKE_C_COMPILER_VERSION "8.50.0")
 set(CMAKE_C_COMPILER_FORCED TRUE)
 set(CMAKE_C_COMPILER_WORKS TRUE)
 set(CMAKE_CXX_COMPILER_ID "${CMAKE_C_COMPILER_ID}")

